jwt-react/src/components/AuthenticatedPingTest/AuthenticatedPingTest.js

```diff
@@ -0,0 +1,32 @@
+import React, { useContext, useState } from 'react';
+import { UserContext } from '../../contexts/userContext.js'
+import { ping } from '../../api/rest.js'
+
+function AuthenticatedPingTest() {
+  const [pingResult, setPingResult] = useState();
+  const {user, setUser, isUserLoggedIn} = useContext(UserContext)
+
+  const testPing = () => {
+    ping()
+    .then((data)=>{
+      setPingResult(data.id)
+    }).catch((error)=> {
+      setUser(null);
+    });
+  }
+
+  return (<div>
+      {isUserLoggedIn() ?
+        <React.Fragment>
+          <h2>Authenticated Request Test</h2>
+          <button onClick={testPing}>Test Ping</button>
+          <p>Ping Result: {pingResult} </p>
```

Comment:
`document.createTextNode()`?

Reply:
I think JSX is doing this under the hood. 

As per: https://reactjs.org/docs/introducing-jsx.html#jsx-prevents-injection-attacks

> By default, React DOM escapes any values embedded in JSX before rendering them. Thus it ensures that you can never inject anything that’s not explicitly written in your application. Everything is converted to a string before being rendered. This helps prevent XSS (cross-site-scripting) attacks.

I'll definitely take a deeper look at this as I think this might suffice, but definitely open to more discussion.

Reply:
Oh wow that's nice to know. I think this comment be be resolved.